packages/expect/src/jest-expect.ts

```diff
@@ -164,17 +164,18 @@ export const JestChaiExpect: ChaiPlugin = (chai, utils) => {
     const pass = jestEquals(actual, expected, [...customTesters, iterableEquality, subsetEquality])
     const isNot = utils.flag(this, 'negate') as boolean
     const { subset: actualSubset, stripped } = getObjectSubset(actual, expected)
-    const msg = utils.getMessage(
-      this,
-      [
-        pass,
-        'expected #{this} to match object #{exp}',
-        'expected #{this} to not match object #{exp}',
-        expected,
-        actualSubset,
-      ],
-    )
     if ((pass && isNot) || (!pass && !isNot)) {
+      const msg = utils.getMessage(
```

Comment:
I haven't follow up this properly, but in some cases, `utils.getMessage` also gets in loop (stacktrace shows `loupe` is doing something).

For now, I hid it inside failure case since `msg` is not used fro success case anyways.